libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -724,10 +724,30 @@ void CodeTransform::visitExpression(Expression const& _expression)
 
 void CodeTransform::visitStatements(vector<Statement> const& _statements)
 {
+	auto fdSeries = groupFunctionDefinitions(_statements);
```

Comment:
Do you really need such a complicated analysis beforehand?
What about something like this:
```
void CodeTransform::visitStatements(vector<Statement> const& _statements)
{
  boost::optional<AbstractAssembly::LabelID> jumpTarget;
  for (auto const& statement: _statements)
  {
    if (statement is a function definition && !jumpTarget)
    {
      jumpTarget = m_assembly.newLabelId();
      m_assembly.appendJumpTo(*jumpTarget, 0);
    }
    if (statement is not a function definition && jumpTarget)
    {
      m_assembly.appendLabel(*jumpTarget)
      jumpTarget = boost::none;
    }
    freeUnusedVariables();
    boost::apply_visitor(*this, statement);
  }
  freeUnusedVariables();
  if jumpTarget)
    m_assembly.appendLabel(*jumpTarget)
}
```

Reply:
Yes the analysis is needed. We don't want to add the `JUMP` statements around single functions. So something like this is needed:

			// create mapping only for series longer than 1
			if (seriesLength > 1)
				fdSeries[seriesStart] = lastFd;

Reply:
Why not add it around single functions? If you always add it here, you can remove it from the code generation of FunctionDefinition.

Reply:
Conditionally removing it from code generation of FunctionDefinition, caused the StackHeight errors (as we had discussed on chat). Unconditionally removing them will break logic for non-single functions.

Reply:
Why does it break the logic for non-single functions?

Reply:
Yes you're right, it won't break for non-single functions.

But I just tested removing the code from FunctionDefinition and the `stack height mismatch` errors are back